### DISCUSSION TITLE
how to use envelop with nestjs?

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [dotansimha/envelop#1554] [Q&A] [answered]
Let's look at the example from the docs real quick
```ts
GraphQLModule.forRoot<ApolloDriverConfig>({
  driver: ApolloDriver,
  typePaths: ['./**/*.graphql'],
  definitions: {
    path: join(process.cwd(), 'src/graphql.ts'),
    outputAs: 'class',
  },
}),
```
So as you can see, the Module takes care of merging all relevant graphql files together to build a schema, so only after this is done you have a schema. Where can I place envelop to be able to access the schema and where to add the wrapper for the requests then?
Seems a bit like a chicken - egg problem.

**Comment by kodeine** [comment, marked as answer]
Looking for the same answer

**Reply by sschneider-ihre-pvs** [reply]
Oh I forgot about this totally. I guess this should work. This is for the latest envelop version.
```ts
import { envelop, useEngine } from '@envelop/core';
import * as GraphQLJS from 'graphql';

export const getEnveloped = envelop({
  plugins: [
    useEngine(GraphQLJS)
  ]
});

@Module({
  imports: [
    GraphQLModule.forRootAsync({
      driver: <driver>,
      useFactory: () => {
        return {
          ...
          executor: async ({ document, operationName, request, schema, context }) => {
            const { execute, contextFactory } = getEnveloped({ req: request, ...context });
            const contextValue = await contextFactory();
            return execute({
              schema,
              document,
              contextValue,
              variableValues: request.variables,
              operationName,
            });
          },
        };
      },
    }),
...
})
export class AppModule {}

```

**Reply by Urigo** [reply]
@kodeine @sschneider-ihre-pvs we have this one: https://github.com/charlypoly/graphql-yoga-nestjs
and we are looking for people to help us lead that project.
If you want to use it in production, we would love to chat and see how we can work on that together, you can lead it and The Guild would support you

**Reply by sschneider-ihre-pvs** [reply]
> @kodeine @sschneider-ihre-pvs we have this one: https://github.com/charlypoly/graphql-yoga-nestjs and we are looking for people to help us lead that project. If you want to use it in production, we would love to chat and see how we can work on that together, you can lead it and The Guild would support you

Very generous you asked, but I think @charlypoly seems to be the right guy for this. I don't have a clue about nestjs drivers and such, it was mere a trial and error to find a way to add envelop and not mutilate the whole context object.

**Reply by Urigo** [reply]
@charlypoly is from The Guild but none of us, including him is using NestJS on a day to day basis.
That's why we all, including @charlypoly , are looking for someone from the NestJS community step up and a user and help us work on it and making it best for NestJS developers

**Reply by hinogi** [reply]
I would love to but I am actually more like a frontend guy and I almost do not know anything about the innards of nestjs. And I also do not use yoga.

**Reply by hinogi** [reply]
I would love to help @charlypoly if I can because I also really like The Guild. Thanks to Felicia Day who gave me that pavlov reflex :D

**Reply by Urigo** [reply]
@sschneider-ihre-pvs @hinogi @kodeine We are all up to date!
https://the-guild.dev/blog/graphql-yoga-nestjs-v9
